app/models/invoice.rb

```diff
@@ -120,6 +120,14 @@ def subtotal_before_prepaid_credits
     amount + wallet_transaction_amount
   end
 
+  def issued_at
+    issuing_time = created_at.in_time_zone(customer.applicable_timezone)
+
+    return issuing_time unless grace_period?
+
+    issuing_time + customer.applicable_invoice_grace_period.days
```

Comment:
This won't work if a customer manually finalizes the invoice before the end of the grace period.
In that case, issuing time should not be equal to this calculation